pkg/reconciler/pipelinerun/resources/pipelinerunstate.go

```diff
@@ -137,29 +132,53 @@ func (state PipelineRunState) checkTasksDone(d *dag.Graph) bool {
 	return true
 }
 
+func (facts *PipelineRunFacts) CheckDAGTasksDone() bool {
```

Comment:
nit: these three public funcs could use comments.  Could be helpful for new devs not yet familiar with the distinction between Tasks and FinalTasks